### DISCUSSION TITLE
Publish schema json to public url

### DISCUSSION THREAD

**Discussion by svrooij** [frenck/action-app-linter#29] [Ideas]
I also have some json schema files for [my pet project](https://svrooij.io/sonos-api-docs/#manual-documentation-file) I publish them to github pages (with custom domain but that doesn't matter).
So my [schema file](https://svrooij.io/sonos-api-docs/schema/documentation.json) is available from the internet and that means they automatically work in VSCode.

If you would clone sonos-api-docs and edit [this file](https://github.com/svrooij/sonos-api-docs/blob/main/docs/documentation.json) it does the json checking right in the editor. Because I've set the `$schema` in the json.

https://github.com/svrooij/sonos-api-docs/blob/d1a95207f37146f6b49cba3b5ce796f67dcd2fa8/docs/documentation.json#L1-L2

You can still use your action, which is great for checking, but this small step would allow all addon developers to use the schema files and have much less faults in their addon configuration.

**Comment by frenck** [comment]
So, the file is public?
<https://raw.githubusercontent.com/frenck/action-addon-linter/main/src/config.schema.json>

**Reply by svrooij** [reply]
I always though that githubusercontent didn't send the correct mimetype. Not sure if vscode takes json schemas that are send without `application/json` content type.

but even if it works, the main documentation of writing addons could state the following.

> Add this schema to your addon build config json file, so you get auto checking the file in vscode while you develop the addon.

**Reply by frenck** [reply]
It doesn't provide those headers indeed.
🤷 

While a nice addition maybe, I don't plan on using it myself that way. If someone wants to make that effort to keep it updated and published, sure 👍

**Reply by svrooij** [reply]
I was only trying to help other developers with an even nicer developer experience ;)
